internal/commands/pull/inputs.go

```diff
@@ -107,5 +115,72 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 		return realm.App{}, err
 	}
 
+	// TODO(REALMC-9462): remove this once /apps has "template_id" in the payload
+	app, err = clients.Realm.FindApp(app.GroupID, app.ID)
+	if err != nil {
+		return realm.App{}, err
+	}
+
 	return app, nil
 }
+
+func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+
+	if i.TemplateID != "" {
+		for _, template := range compatibleTemplates {
+			if template.ID == i.TemplateID {
+				templateZip, ok, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+				if err != nil {
+					return nil, err
+				} else if !ok {
+					return nil, fmt.Errorf("template '%s' does not have a frontend to pull", template.ID)
+				}
+				return map[string]*zip.Reader{template.ID: templateZip}, nil
+			}
+		}
+		return nil, fmt.Errorf("template '%s' is not compatible with this app", i.TemplateID)
+	}
+
+	if len(compatibleTemplates) == 0 {
+		return nil, nil
+	}
+
+	if proceed, err := ui.Confirm("Would you like to export with a template?"); err != nil {
+		return nil, err
+	} else if !proceed {
+		return nil, nil
+	}
+
+	nameOptions := make([]string, len(compatibleTemplates))
```

Comment:
we should just file a followup ticket (and add the TODO here) but this behavior should probably align with `app create`, where instead of prompting the user above ("Would you like to export...") we otherwise just present them a list of templates, with `No template` (confirm the formatting on this) being the first option

Reply:
Added to this ticket 
https://jira.mongodb.org/browse/REALMC-9474